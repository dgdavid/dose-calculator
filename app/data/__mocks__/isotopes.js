
const isotopes = {
  i1: {
    id: 'i1',
    name: 'Isotope 1',
    halfLife: 192.48,
    lambda: 0.0036,
  },
  i2: {
    id: 'i2',
    name: 'Isotope 2',
    halfLife: 6.01,
    lambda: 0.1153,
  },
};

export default isotopes;
