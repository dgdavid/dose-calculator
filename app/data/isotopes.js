const isotopes = {
  'iodine-131': {
    id: 'iodine-131',
    name: 'Iodine 131',
    halfLife: 192.48,
    lambda: 0.0036,
  },
  'technetium-99m': {
    id: 'technetium-99m',
    name: 'Technetium 99m',
    halfLife: 6.01,
    lambda: 0.1153,
  },
};

export default isotopes;
