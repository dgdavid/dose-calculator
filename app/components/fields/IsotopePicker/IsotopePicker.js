import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Picker, Text, View } from 'react-native';

import isotopes from '../../../data/isotopes';
import colors from '../../../config/colors';

const IsotopePickerOptions = Object.keys(isotopes).map((key) => (
  <Picker.Item
    label={isotopes[key].name}
    key={key}
    value={key}
  />
));

const IsotopePicker = ({ prompt, ...props }) => (
  <Fragment>
    <View style={styles.wrapper}>
      <Text style={styles.prompt}>{prompt}</Text>
      <Picker
        style={styles.picker}
        {...props}
      >
        {IsotopePickerOptions}
      </Picker>
    </View>
  </Fragment>
);

IsotopePicker.propTypes = {
  prompt: PropTypes.string,
};

IsotopePicker.defaultProps = {
  prompt: 'Select isotope',
};

const styles = {
  wrapper: {
    // Fix to get Picker clipped at the edges of wrapper and make the "prompt trick" works.
    // See https://github.com/facebook/react-native/issues/7817#issuecomment-264851951 to know more.
    borderWidth: 0,
  },

  prompt: {
    color: colors.dimGray,
    fontSize: 30,
    paddingTop: 5,
    alignSelf: 'center',
  },

  picker: {
    position: 'absolute',
    top: 0,
    width: 1000,
    height: 1000,
  },
};

export default IsotopePicker;
