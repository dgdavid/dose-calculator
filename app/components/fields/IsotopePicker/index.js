import IsotopePicker from './IsotopePicker';

export default IsotopePicker;
