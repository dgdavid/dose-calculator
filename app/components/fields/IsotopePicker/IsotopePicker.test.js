import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import IsotopePicker from './IsotopePicker';
import isotopes from '../../../data/isotopes';

jest.mock('../../../data/isotopes');

test('renders correctly', () => {

  const tree = renderer.create(<IsotopePicker isotope={isotopes.i2} />).toJSON();

  expect(tree).toMatchSnapshot();
});
