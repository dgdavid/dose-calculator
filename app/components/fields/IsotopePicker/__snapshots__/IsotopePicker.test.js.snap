// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderWidth": 0,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#696969",
        "fontSize": 30,
        "paddingTop": 5,
      }
    }
  >
    Select isotope
  </Text>
  <View
    style={
      Object {
        "height": 1000,
        "position": "absolute",
        "top": 0,
        "width": 1000,
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Technetium 99m",
            "textColor": undefined,
            "value": "technetium-99m",
          },
          Object {
            "label": "Iodine 131",
            "textColor": undefined,
            "value": "iodine-131",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
